### DISCUSSION TITLE
What you *can* and *cannot* do with PHP in patterns

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#210] [Topic Ideas]
A solid tutorial covering what functions are available to you when building patterns. For example, did you know that patterns are executed on `init`, a time when most WP conditionals are not yet set?

Why use PHP? Internationalization, dynamic URLs for loading media, and more.

You can also do things like loops instead of printing the block markup for repeated sections of the pattern with the same content (for example, a Columns block with the same nested columns).

**Comment by bph** [comment]
After this topic was approved, the discussion is locked. 
Conversation continues on the issue

- #214

**Comment by bph** [comment]
Will be covered in the Theme Handbook ([see closing comment on the issue](https://github.com/WordPress/developer-blog-content/issues/214#issuecomment-1986596988)) which also closed this discussion
